BEIMA.Backend/ReportService/ReportWriter.cs

```diff
@@ -0,0 +1,286 @@
+﻿using BEIMA.Backend.MongoService;
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization;
+using MongoDB.Driver;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BEIMA.Backend.ReportService
+{
+    /// <summary>
+    /// This class abstracts basic report generation operations. It is implemented as a 
+    /// singleton class. 
+    /// </summary>
+    public class ReportWriter : IReportService
```

Comment:
Having looked through this now, I think what you said about making this a static class sounds like it could be a good idea here.  My suggestion here however would be to have the devices/device types be passed in, as opposed to getting them from the database directly, and to do all the database stuff when we implement the endpoint for this. This way we don't have to make any mock databases when testing this. As it is right now, all of our database calls are on the endpoint classes themselves and I think it would be a good idea to make sure that we keep it that way to reduce dependencies.

Reply:
If we use a static class for this instead, we also don't need to use the interface/ReportDefinition classes, as we would be able to call this class directly. We only need those if we want to be able to mock this class in testing. However, because we aren't interacting with anything outside the program here (if we decide to move out the database stuff), there isn't really a reason why this would need to be mocked.

Reply:
Yeah I think it would also help remove the db dependency because it feels weird for this to be a service that relies on another service.